src/main/webapp/MillenniaDev_script.js

```diff
@@ -0,0 +1,122 @@
+//Returns a list of databases from Cloud Spanner as html dropdown
+function getDatabases(){
+  fetch("/databases").then(response => response.json()).then((list) => {
+    let dropdown = document.getElementById('list-databases');
+    dropdown.options.remove(0);
+    let defaultOption = document.createElement('option');
+    defaultOption.text = 'Select a Database';
+
+    dropdown.add(defaultOption);
+    dropdown.selectedIndex = 0;
+
+    let option;
+    
+    for (let i in list) {
+      option = document.createElement('option');
+      option.text = list[i];
+      option.value = list[i];
+      if(list[i] != null){
+        dropdown.add(option);
+      } 
+    }
+  });
+}
+
+//Checks whether the user has left both of the user_id/device_id input box empty. If true, the user cannot submit the filter
+// query,if false submit
+function isFilterInputEmpty(){
+    var userID = document.getElementById("user_id").value;
+    var deviceID = document.getElementById("device_id").value;
+
+    //Return a message saying that the filters cannot be applied unless at least one of the inputs is filled.
+    if(userID == "" && deviceID == ""){
+        alert("Applying filters failed: Please input either user_id, device_id or both.");
+        return false;
+    }
+    return true;
+}
+
+function showFiltersPanel(){
+  var x = document.getElementById("filter-box");
+  if (x.style.display === "none") {
+    x.style.display = "block";
+  } else {
+    x.style.display = "none";
+  }
+}
+
+//Returns checkbox dropdowns of the selected table's columns which can then be filtered
+function filterColumns() {
```

Comment:
This function is really long -- try to split it up into smaller functions if possible! Each smaller function could have its own specific purpose.

Reply:
I can try the reason all the styling and creating of elements is within one function is because they're dynamically created with the DOM, so they're not permanently on the html to be able to style them outside of js with a css file, because an error will come up with the elements as null. I'll do a bit more research into checking if it's possible to safely split this function up!